samples/Echo_Ssl/app/application.cpp

```diff
@@ -109,19 +109,24 @@ bool onReceive(TcpClient& tcpClient, char *data, int size) {
 	return tcpClient.send(data, size);
 }
 
-void connectOk()
+void connectOk(String ssid, uint8_t ssid_len, uint8_t bssid[6], uint8_t channel)
 {
-	debugf("Connected. Got IP: %s", WifiStation.getIP().toString().c_str());
-
-	client = new TcpClient(TcpClientDataDelegate(onReceive));
-	client->addSslOptions(SSL_SERVER_VERIFY_LATER);
-	client->connect(IPAddress(SERVER_IP), 4444, true);
+	debugf("Connected.");
 }
 
-void connectFail()
+void connectFail(String ssid, uint8_t ssid_len, uint8_t bssid[6], uint8_t reason)
 {
 	debugf("I'm NOT CONNECTED!");
-	WifiStation.waitConnection(connectOk, 10, connectFail); // Repeat and check again
+	WifiStation.disconnect();
```

Comment:
This, disconnect/connect was my fault, I'm copy/paste it from my application Class and forgot to remove it completely. IMHO we should do nothing, but some debug message in STADisconnect-equivalent event handler. In my app class I start access point (if not started already) to enable user change ssid/pwd if no connection to predefined ssid but in generic there is no means to do so.